Assets/HoloToolkit/Input/Scripts/Cursor/Cursor.cs

```diff
@@ -0,0 +1,478 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// Object that represents a cursor in 3D space controlled by gaze.
+    /// </summary>
+    public abstract class Cursor : MonoBehaviour, ICursor
+    {
+        /// <summary>
+        /// Enum for current cursor state
+        /// </summary>
+        public enum CursorStateEnum
+        {
+            /// <summary>
+            /// Useful for releasing external override.
+            /// See <c>CursorStateEnum.Contextual</c>
+            /// </summary>
+            None = -1,
+            /// <summary>
+            /// Not IsHandVisible
+            /// </summary>
+            Observe, 
+            /// <summary>
+            /// Not IsHandVisible AND not IsInputSourceDown AND TargetedObject exists
+            /// </summary>
+            ObserveHover, 
+            /// <summary>
+            /// IsHandVisible AND not IsInputSourceDown AND TargetedObject is NULL
+            /// </summary>
+            Interact,
```

Comment:
I polled a couple others and they also said "Ready" would be good to swap out for "Interact" as well.  My comment was toward other input devices not just gesture and what term is more applicable to cursor and input UX design.  What does "Ready" mean when using 3DoF input from the clicker?  Food for thought.  Good feedback, anything that makes more sense from broader perspective I'm all for.